toutv/bos.py

```diff
@@ -58,9 +69,19 @@ def get_proxies(self):
 
     def _do_request(self, url, timeout=None, params=None):
         proxies = self.get_proxies()
+        auth = self.get_auth()
 
         try:
-            r = requests.get(url, params=params, headers=toutv.config.HEADERS,
+            headers = dict(toutv.config.HEADERS)
+
+            if auth and params:
```

Comment:
why "auth and param"?

What happens if no params are passed, but we still want to authenticate?